op-challenger/fault/loader.go

```diff
@@ -0,0 +1,100 @@
+package fault
+
+import (
+	"context"
+	"math/big"
+
+	"github.com/ethereum/go-ethereum/accounts/abi/bind"
+	"github.com/ethereum/go-ethereum/log"
+)
+
+// ClaimFetcher is a minimal interface around [bindings.FaultDisputeGameCaller].
+// This needs to be updated if the [bindings.FaultDisputeGameCaller] interface changes.
+type ClaimFetcher interface {
+	ClaimData(opts *bind.CallOpts, arg0 *big.Int) (struct {
+		ParentIndex uint32
+		Countered   bool
+		Claim       [32]byte
+		Position    *big.Int
+		Clock       *big.Int
+	}, error)
+	ClaimDataLen(opts *bind.CallOpts) (*big.Int, error)
+}
+
+// loader pulls in fault dispute game claim data periodically and over subscriptions.
+type loader struct {
+	log log.Logger
+
+	state Game
+
+	claimFetcher ClaimFetcher
+}
+
+// NewLoader creates a new [loader].
+func NewLoader(log log.Logger, state Game, claimFetcher ClaimFetcher) *loader {
+	return &loader{
+		log:          log,
+		state:        state,
+		claimFetcher: claimFetcher,
+	}
+}
+
+// fetchClaim fetches a single [Claim] with a hydrated parent.
+func (l *loader) fetchClaim(ctx context.Context, arrIndex uint64) (*Claim, error) {
```

Comment:
`ctx` isn't actually being used so can be removed from the API and from `FetchClaims`.

I'd probably modify it to just return `Claim` rather than a pointer as well given the only usage immediately dereferences it.  Just means returning `Claim{}, err` in a few places but that's fine.

Reply:
Because this code does network calls, I think we should add the `ctx` to `CallOpts` (which is how it is uses in this API) instead of removing it.